clap_complete/src/dynamic/completer.rs

```diff
@@ -339,6 +352,22 @@ fn complete_arg_value(
     let mut values = Vec::new();
     debug!("complete_arg_value: arg={arg:?}, value={value:?}");
 
+    let (prefix, value) = match arg.get_value_delimiter() {
+        Some(delimiter) => match value {
+            Ok(value_os) => match value_os.rfind(delimiter) {
+                Some(pos) => {
+                    let (prefix, value) = value_os.split_at(pos);
+                    let prefix = &prefix[..prefix.len()];
+                    let value = &value[1..];
```

Comment:
I assume the `1` is for the delimiter.  We should check the length of the delimiter and split after that

Reply:
Since the delimiter should only be a `char`, why are we checking the length of the delimiter?

Reply:
`char`s are basically a `u32` and, when converted to `&str`, can take a variable number of bytes, e.g. see https://doc.rust-lang.org/std/primitive.char.html#method.encode_utf8